src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -33,21 +33,29 @@ class HttpUtils
     private $secureDomainRegexp;
 
     /**
-     * @param UrlGeneratorInterface                       $urlGenerator       A UrlGeneratorInterface instance
-     * @param UrlMatcherInterface|RequestMatcherInterface $urlMatcher         The URL or Request matcher
-     * @param string|null                                 $domainRegexp       A regexp the target of HTTP redirections must match, scheme included
-     * @param string|null                                 $secureDomainRegexp A regexp the target of HTTP redirections must match when the scheme is "https"
+     * @param UrlGeneratorInterface                       $urlGenerator A UrlGeneratorInterface instance
+     * @param UrlMatcherInterface|RequestMatcherInterface $urlMatcher   The URL or Request matcher
+     * @param string                                      $cookieSecure Cookie secure configration option
+     * @param string                                      $domainRegexp
      *
      * @throws \InvalidArgumentException
      */
-    public function __construct(UrlGeneratorInterface $urlGenerator = null, $urlMatcher = null, string $domainRegexp = null, string $secureDomainRegexp = null)
+    public function __construct(UrlGeneratorInterface $urlGenerator = null, $urlMatcher = null, string $cookieSecure, string $domainRegexp)
```

Comment:
that's a BC break.
Instead, I'd suggest to add a static factory method to handle the runtime creation of instances

Reply:
Ok will do thxs.

Reply:
@nicolas-grekas Please check the last commit, is that what you were sugesting?

Reply:
yes! I didn't look at the details yet, but that's the approach I was suggesting indeed.